src/Doctrine/Orm/Filter/FilterExpressionProvider.php

```diff
@@ -96,6 +112,29 @@ protected static function applyFilter(
         $expression = $queryBuilder->getDQLPart('where');
         $queryBuilder->where($originalWhere);
 
+        if ($innerJoinsLeft) {
+            $joinPart = $queryBuilder->getDQLPart('join');
+            $result = [];
+            foreach ($joinPart as $rootAlias => $joins) {
+                /** @var Join $joinExp */
+                foreach ($joins as $i => $joinExp) {
+                    if ($joinExp->getJoinType() === Join::INNER_JOIN) {
+                        $result[$rootAlias][$i] = new Join(
+                            Join::LEFT_JOIN,
+                            $joinExp->getJoin(),
+                            $joinExp->getAlias(),
+                            $joinExp->getConditionType(),
+                            $joinExp->getCondition(),
+                            $joinExp->getIndexBy()
+                        );
+                    } else {
+                        $result[$rootAlias][$i] = $joinExp;
+                    }
+                }
+            }
+            $queryBuilder->add('join', $result);
```

Comment:
Could you can extract this in his own function, in the same way that the metaclass-nl package?